File: features/operational/contact_us/src/main/java/com/tokopedia/contactus/inboxticket2/view/fragment/ServicePrioritiesBottomSheet.java
package com.tokopedia.contactus.inboxticket2.view.fragment;

import android.content.Context;
import android.graphics.Color;
import android.support.annotation.NonNull;
import android.text.SpannableString;
import android.text.Spanned;
import android.text.TextPaint;
import android.text.method.LinkMovementMethod;
import android.text.style.ClickableSpan;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.TextView;

import com.tokopedia.contactus.ContactUsModuleRouter;
import com.tokopedia.contactus.R;



public class ServicePrioritiesBottomSheet extends FrameLayout implements View.OnClickListener {

    public static final String LEARN_MORE_TEXT = "Pelajari Selengkapnya";
    private Context mContext;
    private final CloseServicePrioritiesBottomSheet closeServicePrioritiesBottomSheet;

    public ServicePrioritiesBottomSheet(@NonNull Context context, CloseServicePrioritiesBottomSheet closeServicePrioritiesBottomSheet) {
        super(context);
        mContext = context;
        this.closeServicePrioritiesBottomSheet = closeServicePrioritiesBottomSheet;
        init();
    }

    private void init() {

        View helpfullView = LayoutInflater.from(getContext()).inflate(R.layout.service_priorities_bottom_sheet_layout, this, true);
        TextView link  = helpfullView.findViewById(R.id.tv_service_priorities_officialstore_desc);
        TextView close = helpfullView.findViewById(R.id.txt_close);
        String text = mContext.getString(R.string.service_priorities_officialstore_desc);
        SpannableString spannableString = new SpannableString(text);
        int startIndexOfLink = text.indexOf(LEARN_MORE_TEXT);
        spannableString.setSpan(new ClickableSpan() {
            @Override
            public void onClick(View view) {
                mContext.startActivity(((ContactUsModuleRouter) (mContext.getApplicationContext()))
                        .getWebviewActivityWithIntent(mContext,
                                mContext.getString(R.string.learn_more_link)));
            }

            @Override
            public void updateDrawState(TextPaint ds) {
                super.updateDrawState(ds);
                ds.setUnderlineText(false);
                ds.setColor(getResources().getColor(R.color.green_250)); // specific color for this link
            }
        }, startIndexOfLink, startIndexOfLink + LEARN_MORE_TEXT.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
        link.setHighlightColor(Color.TRANSPARENT);
        link.setMovementMethod(LinkMovementMethod.getInstance());

        link.setText(spannableString, TextView.BufferType.SPANNABLE);

        close.setOnClickListener(this);
    }
    @Override
    public void onClick(View view) {
        closeServicePrioritiesBottomSheet.onClickClose();
    }

    public interface CloseServicePrioritiesBottomSheet {
        void onClickClose();
    }
}

File: features/operational/contact_us/src/main/java/com/tokopedia/contactus/inboxticket2/view/fragment/ServicePrioritiesBottomSheet.kt
package com.tokopedia.contactus.inboxticket2.view.fragment

import android.content.Context
import android.graphics.Color
import android.text.SpannableString
import android.text.Spanned
import android.text.TextPaint
import android.text.method.LinkMovementMethod
import android.text.style.ClickableSpan
import android.view.View
import android.widget.TextView
import com.tokopedia.abstraction.common.utils.view.MethodChecker
import com.tokopedia.applink.RouteManager
import com.tokopedia.contactus.R
import com.tokopedia.unifycomponents.BottomSheetUnify

class ServicePrioritiesBottomSheet(private val mContext: Context,
                                   private val closeServicePrioritiesBottomSheet: CloseServicePrioritiesBottomSheet) : BottomSheetUnify(), View.OnClickListener {
    init {
        val priorityBottomSheetview = View.inflate(mContext, R.layout.service_priorities_bottom_sheet_layout, null)
        val link = priorityBottomSheetview.findViewById<TextView>(R.id.tv_service_priorities_officialstore_desc)
        val close = priorityBottomSheetview.findViewById<TextView>(R.id.txt_close)
        val text = mContext.getString(R.string.service_priorities_officialstore_desc)
        val spannableString = SpannableString(text)
        val startIndexOfLink = text.indexOf(LEARN_MORE_TEXT)
        spannableString.setSpan(object : ClickableSpan() {
            override fun onClick(view: View) {
                mContext.startActivity(RouteManager.getIntent(mContext, mContext.getString(R.string.learn_more_link)))
            }

            override fun updateDrawState(ds: TextPaint) {
                super.updateDrawState(ds)
                ds.isUnderlineText = false
                ds.color = MethodChecker.getColor(mContext, R.color.contact_us_green_250) // specific color for this link
            }
        }, startIndexOfLink, startIndexOfLink + LEARN_MORE_TEXT.length, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
        link.highlightColor = Color.TRANSPARENT
        link.movementMethod = LinkMovementMethod.getInstance()
        link.setText(spannableString, TextView.BufferType.SPANNABLE)
        close.setOnClickListener(this)

        setChild(priorityBottomSheetview)
        showKnob= false
        showCloseIcon = false
        showHeader = false
    }

    override fun onClick(view: View) {
        closeServicePrioritiesBottomSheet.onClickClose()
    }

    interface CloseServicePrioritiesBottomSheet {
        fun onClickClose()
    }

    companion object {
        const val LEARN_MORE_TEXT = "Pelajari Selengkapnya"
    }

}